### PR TITLE
Converts loadouts, limbs, and languages to preference middleware

### PR DESCRIPTION
**Pull request by MrMelbert**
They in tabs now. 

image

### CODE REVIEW

**Review by RimiNosha — COMMENTED** [review]
On my end, I've been able to get rid of just about all the `name` fields on the item datums by using `style={{ 'text-transform': 'capitalize' }}` on `_LoadoutManger.tsx:147` along with using initial item names. It so far plays well with everything, including the weirder names like Nar'Sie.

Hopefully the removal of the need to type out a million item names will make adding new items less tedious.

You'll need to edit the category datum to check just the item_type instead and not throw an angry message, though, but no further edits are required.

File: maplestation_modules/code/modules/loadouts/loadout_ui/loadout_manager.dm [hunk 1]
```diff
@@ -68,113 +73,86 @@
...
-	var/save_on_close = FALSE
 
-/datum/loadout_manager/New(user)
-	owner = CLIENT_FROM_VAR(user)
-	owner.open_loadout_ui = src
-	loadout = owner.prefs.read_preference(/datum/preference/loadout)
+/datum/preference_middleware/loadout/New(datum/preferences)
+	. = ..()
+
+	loadout = src.preferences.read_preference(/datum/preference/loadout)
```
Comment (RimiNosha) [inline maplestation_modules/code/modules/loadouts/loadout_ui/loadout_manager.dm:82]: This is called before preferences are properly initialized, also is called with null preferences due to how static caching works.

I'd also just recommend getting rid of the `loadout` var, and instead read from prefs when you need the list inside procs. It's cached already anyways, and you're already passing prefs via `manager` where you'd need this.

Or if you really want to be as cool on the cycles as possible, attach this to the preferences datum, and handle it from inside there. Doing all this inside middleware won't work without monmod edits.

**Comment by MrMelbert** [comment]
> Hopefully the removal of the need to type out a million item names will make adding new items less tedious.

With hindsight the name should default to the initial item name if left unset. Makes it less tedious (as you say) but still allows for humans to edit certain item names if they want to be more specific.

File: maplestation_modules/code/modules/loadouts/loadout_ui/loadout_manager.dm (hunk 1 above)
Comment (MrMelbert) [reply maplestation_modules/code/modules/loadouts/loadout_ui/loadout_manager.dm:82]: Ah yes that does explain the runtimes I saw but had not bother to fix yet. Good shout.

File: maplestation_modules/code/modules/client/preferences/languages.dm [hunk 2]
```diff
@@ -82,122 +77,94 @@
...
 /datum/language/piratespeak
 	available_as_pref = TRUE
 
 /datum/language/yangyu
 	available_as_pref = TRUE
+	banned_from_species = /datum/species/ornithid
+
+/datum/language/shadowtongue
+	available_as_pref = TRUE
+
```
Comment (Jolly-66) [inline maplestation_modules/code/modules/client/preferences/languages.dm:119]:
```suggestion

```

You have an accidental extra space here.

**Comment by MrMelbert** [comment]
I think this is ready, if not, we'll fix it in post

**Pull request merged**